### PR TITLE
Exclude archived repositories

### PR DESCRIPTION
**Pull request by adambraimbridge**
Resolves https://github.com/Financial-Times/tako/issues/46

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -32,13 +32,18 @@ module.exports = async app => {
 		const added = context.payload.repositories_added;
 
 		added.forEach(async repository => {
-			// Save this new repository.
-			repositoryStore.set(repository.id, {
-				id: repository.id,
-				name: repository.name
-			});
+			if (repository.archived) {
```
Comment (sjparkinson) [inline src/index.js]: I don't believe this property exists on repositories for this event.

I'll add more detail as a comment to the test.

File: test/initialise.test.js [hunk 2]
```diff
@@ -26,7 +26,10 @@ describe("initalise.js", () => {
 					data: { owner: { login: "Financial-Times" } }
 				}),
 				getInstallationRepositories: jest.fn().mockResolvedValue({
-					data: { repositories: [{ id: 12345, name: "foo-bar" }] }
+					data: { repositories: [
+						{ id: 12345, name: "foo-bar" },
+						{ id: 23456, name: "foo-bar-archived", archived: true },
```
Comment (sjparkinson) [inline test/initialise.test.js]: From https://developer.github.com/v3/activity/events/types/#installationrepositoriesevent, I'm not sure that the `archived` property exists here.

You may need to make a call to `context.github.repo.get({ name, owner })` for more information for each repository.

Or you could make a call to [`installation.apps.getInstallationRepositories`](https://developer.github.com/v3/apps/installations/#list-repositories) like we do when the app starts, as that returns the right information.

File: src/initialise.js [hunk 3]
```diff
@@ -109,8 +109,13 @@ module.exports = async app => {
 		);
 
 		// Save each repository to our global map of repositories.
-		repositories.forEach(({ id, name }) => {
-			repositoryStore.set(id, { id, name });
+		repositories.forEach(({ id, name, archived }) => {
+			if (archived) {
+				logger.info(`Excluding ${name} (because it is )archived.`);
```
Comment (simonplend) [inline src/initialise.js]: Random `(` and `)` in this log message.

Comment (sjparkinson) [reply src/initialise.js]: image

😂

Comment (sjparkinson) [reply src/initialise.js]: image

😂

File: test/initialise.test.js (hunk 2 above)
Comment (sjparkinson) [reply test/initialise.test.js]: To confirm, I tested this out by adding a removing https://github.com/financial-times-sandbox/testing-archived.

image

Comment (adambraimbridge) [reply test/initialise.test.js]: Curses

File: src/initialise.js (hunk 3 above)
Comment (adambraimbridge) [reply src/initialise.js]: (How )bizarre

**Comment by simonplend** [comment]
My concern about the way this currently works is that we will only be able to identify which repositories are archived when the application starts. Ignoring the fact that we restart our Heroku dynos once a day, as this isn't "application" behaviour", in general this means we won't actually end up filtering out archived repositories.

Is it possible to additionally listen for the `RepositoryEvent` type and check if the `action` is `archived`? https://developer.github.com/v3/activity/events/types/#repositoryevent

**Comment by simonplend** [comment]
Answering my own question here: yes!

> The app can listen to any of the GitHub webhook events.

Source: https://probot.github.io/docs/webhooks/

Also, see: https://developer.github.com/webhooks/#events

**Comment by adambraimbridge** [comment]
Yes. In this case I'd need to deal with events for both archiving and unarchiving repositories.

**Comment by adambraimbridge** [comment]
Taking a new approach, so closing this.

**Pull request closed without merging**